deploy/deploy-centralvenv.sh

```diff
@@ -0,0 +1,1000 @@
+      -d  <wmcore_path>              WMCore virtual environment target path to be used for this deployment [Default: ./WMCore.venv3]
+      -p  <patches>                  List of PR numbers [Default: None]
+                                     (in double quotes and space separated e.g. "5906 5934 5922")
+      -m  <security string>          The security string to be used during deployment. Will be needed at startup [Default: ""]
+      -l  <service_list>             List of services to be deployed [Default: "wmcore" - WMCore metapackage]
+                                     The full list of installable services: "reqmgr2 reqmgr2ms workqueue reqmon t0_reqmon"
+                                     (in double quotes and space separated e.g. "rqmgr2 reqmgr2ms")
+                                     (pip based package version syntax is also acceptable e.g. "wmcore==2.0.0")
+      -i <pypi-index>                The pypi index to use (i.e. prod or test) [Default: prod - pointing to https://pypi.org/simple/]
+      -h <help>                      Provides help to the current script
+
+    # Example: Deploy WMCore central services version 2.0.3rc1 linked with 'cmsweb-test1.cern.ch' as a frontend from 'test' pypi index
+    #          at destination /data/tmp/WMCore.venv3/ and using 'Some security string' as a security string for operationss at runtime:
+    # ./deploy-centralvenv.sh -c cmsweb-test1.cern.ch -i test -l wmcore==2.0.3rc1 -d /data/tmp/WMCore.venv3/ -m "Some security string"
+
+    # Example: Same as above, but do not cleanup deployment area and reuse it from previous installtion - usefull for testing behaviour
+    #          of different versions during development or mix running from source and from pypi installed packages.
+    #          NOTE: The 'current' link must point to the proper deployment area e.g. either to 'srv/master' for running from source
+    #                or to 'srv/2.0.3rc1' for running from pypi installed package):
+    # ./deploy-centralvenv.sh -c cmsweb-test1.cern.ch -n -i test -l wmcore==2.0.3rc1 -d /data/tmp/WMCore.venv3/ -m "Some security string"
+
+    # Example: Deploy WMCore central services from source repository, use tag 2.0.0.pre3, linked with 'cmsweb-test1.cern.ch' as a frontend
+    #          at destination /data/tmp/WMCore.venv3/ and using 'Some security string' as a security string for operationss at runtime:
+    # ./deploy-centralvenv.sh -c cmsweb-test1.cern.ch -s -t 2.0.0.pre3 -p "10003 9998" -d /data/tmp/WMCore.venv3/ -m "Some security string"
+
+    # Example: Same as above, but assume 'Yes' to all questions. To be used in order to chose the default flow and rely only
+    #          on the pameters set for configuring the deployment steps. This will avoid human intervention during deployment:
+    # ./deploy-centralvenv.sh -c cmsweb-test1.cern.ch -y -s -t 2.0.0.pre3 -p "10003 9998" -d /data/tmp/WMCore.venv3/ -m "Some security string"
+
+
```

Comment:
You can deploy it on a laptop as it is right now. The FE to be used is still coming from the `service_config` files from gilab and for the time being this parameter plays no role here. We still need to take a decision if to keep it and apply it for the current deployment or to completely get rid of it and rely only on the parameters set at the `service_configs`. Please see my comment above on the topic: https://github.com/dmwm/WMCore/pull/10980#discussion_r859496766

Reply:
Here I asked to provide concrete example how to setup things on laptop w/o central service. As I wrote already in another comment I advise you to allow users to specify their configs. Doing this way I can easily provide my config and run things locally w/o any central service or FE.

Reply:
> Here I asked to provide concrete example how to setup things on laptop w/o central service

Every each of those examples I already provided are capable of executing on a laptop or where else a person would like to. They are not tied with anything, but enough available space at the destination. Anyway, I did provide an example and I have also gotten rid of the mandatory parameter as you asked.

When it comes to `w/o central service` I do not understand what you mean.... This script is the one deploying the central services. Maybe it is due to a misconception of terminology etc. but as it comes to WMCore and my understanding the FE is not the `Central services`  themselves. All services deployed from this script are central services. And as of now, because of the lack of separation between cmsweb deployment procedures and WMCore ones, simply the FE has always been part of central services as well, but it itself is not the whole of it. The rest is basically related to WMAgent components and external services (e.g. databases as a services etc.) which we use.

> Doing this way I can easily provide my config and run things locally w/o any central service or FE

You can already provide your won configs. Just upload them to a gitlab or any repository and point them through the options at deployment time like for the current default `-g https://:@gitlab.cern.ch:8443/cmsweb-k8s/services_config.git -j preprod` or what ever private repository and branch you wish.  Upon that once they are properly distributed at the services deployment area you may continue editing them as usual.